pkg/controllers/controlplanemachinesetgenerator/utils.go

```diff
@@ -193,6 +195,15 @@ func buildFailureDomains(logger logr.Logger, machineSets []machinev1beta1.Machin
 		cpmsFailureDomain = buildAzureFailureDomains(failureDomains)
 	case configv1.GCPPlatformType:
 		cpmsFailureDomain = buildGCPFailureDomains(failureDomains)
+	case configv1.OpenStackPlatformType:
+		cpmsFailureDomain, err = buildOpenStackFailureDomains(failureDomains)
+		if err != nil {
+			return nil, fmt.Errorf("failed to build OpenStack failure domains: %w", err)
+		}
+
+		if cpmsFailureDomain.OpenStack == nil {
```

Comment:
I think we should change this, because again I don't think OpenStack should be the only platform where there is no failure domain.
Although it might involve openshift/api change for other platforms. 

Ideal all the logic would live in the `build<platform>FailureDomains`  functions, here I just wanted to highlight some specifics for OpenStack; which I hope we can make generic.

Reply:
Conceptually, "no failure domain" is exactly the same as "one failure domain". Can we use this fact to our advantage?

Reply:
I'm testing dropping that block in https://github.com/openshift/cluster-control-plane-machine-set-operator/pull/195/commits/e17445dba9c963a55103fae40dc679005b10e2da - I want to see how CI works because in our CI there is no failure domain.

Reply:
I leave this comment open for CCPMSO reviewers to answer @pierreprinetti 's question.

Reply:
This pretty much closes my question. The current spec seems to be internally-consistent and it does **not** allow for a failure domain with only default values (`""`).

Reply:
As a corollary: we must support zero failure domains. [Here's how](https://github.com/openshift/api/pull/1509/files#diff-561b883d6ecc21e96496df809f4b7c0dff93b3d23bf8d9164df87a88fb1a48fcR93-R94):

```yaml
            failureDomains:
              platform: ""
```

@EmilienM can you please make sure that this syntax is allowed and understood here in CPMS, possibly covering it with a test?

Reply:
What do you think of :
```suggestion
		if cpmsFailureDomain.OpenStack == nil || cpmsFailureDomain.Platform == "" {
```

Reply:
for the test, I wonder if we need more than: https://github.com/openshift/cluster-control-plane-machine-set-operator/pull/195/files#diff-edef557ce8770d851bfcb37599dc53c70a09366a05fa9455420f370f6fb9722aR2549

Reply:
I've added it in the third commit, let me know what you think.